docs/design_proposals/channel_config.md

```diff
@@ -0,0 +1,216 @@
+
+```toml
+default_channels = ["bioconda", "conda-forge"]
+default_url = "https://my.custom.mirror/{channel}"
+
+[[channels]]
+name = "conda-forge"
+url = [
+    "https://conda.anaconda.org/conda-forge",
+    "https://repo.prefix.dev/conda-forge"
+]
+
+[[channels]]
+name = "prefix-internal"
+url = "https://repo.prefix.dev/prefix-internal"
+
+[[channels]]
+name = "bioconda"
+url = "https://repo.prefix.dev/bioconda"
+```
+
+!!!note "Configure in `pixi.toml`"
+    All keys specified in `rattler.toml` can also be specified in `pixi.toml`.
+    `rattler.toml` takes precedence over `pixi.toml`.
+
+## Use cases
+
+### Use the same mirror everywhere
+
+If you want to use the same mirror for all channels, you can set the `default_url` key in the configuration file.
+
+```toml title="rattler.toml"
+default_url = "https://my.custom.mirror/{channel}"
+
+[[channels]]
+name = "bioconda"
+url = "https://repo.prefix.dev/bioconda"
+```
+
+This results in all channels being redirected to this mirror.
+
+```toml title="pixi.toml"
+...
+channels = [
+    "conda-forge",
+    "bioconda",
+    "https://conda.anaconda.org/cf-staging",
+    "https://repo.prefix.dev/prefix-internal"
+]
+...
+```
+
+Using this `pixi.toml` with the above `rattler.toml` will result in the following behavior:
+
+- `conda-forge` will be redirected to `https://my.custom.mirror/conda-forge`
+- `bioconda` will be redirected to `https://repo.prefix.dev/bioconda` since it has higher priority than `default_mirrors`
+- `https://conda.anaconda.org/cf-staging` and `https://repo.prefix.dev/prefix-internal` will not being redirected since they specify a full URL
+
+!!!note "URLs in `channels`"
+    If you specify a full URL in `channels`, it will never be changed by `rattler.toml`. Only named channels will be redirected.
+
+!!!tip "Use `prefix.dev` mirrors"
+    If you want to use the mirrors hosted on [prefix.dev](https://prefix.dev), you can use the following configuration:
+
+    ```toml title="rattler.toml"
+    default_url = "https://repo.prefix.dev/{channel}"
+    ```
+
+### Default channels
+
+```toml title="rattler.toml"
+default_channels = ["my-private-channel", "bioconda", "conda-forge"]
+
+[[channels]]
+name = "my-private-channel"
+url = "https://my.private.quetz/get/my-private-channel"
+```
+
+This will result in the channels `bioconda` and `conda-forge` and `my-private-channel` being used when you run `pixi init` or `pixi global install` and `rattler-build` (assuming you don't override the channels using `-c`).
+
+`pixi init` would create the following `pixi.toml`:
+
+```toml title="pixi.toml"
+# ...
+channels = [
+    "my-private-channel",
+    "bioconda",
+    "conda-forge"
+]
+# ...
+
+[[channels]]
+name = "my-private-channel"
+url = "https://my.private.quetz/get/my-private-channel"
+```
+
+`pixi init --ignore-rattler-config` would create the following `pixi.toml`:
+
+```toml title="pixi.toml"
+# ...
+channels = ["conda-forge"]
+# ...
+```
+
+### Multiple mirrors
```

Comment:
I will probably not use this feature and I'm wondering how many people will... Is the performance difference that large?
In all corporate environments that I've worked in so far, you have a single mirror and in the outside world I probably just wouldn't bother setting up multiple mirrors and just use the default one.

If we don't support multiple mirrors for the same channel, it would make the config a bit easier to understand.

Reply:
I also think multiple mirrors is a pretty rare use case and makes things a bit harder to understand, but I'm still on the fence about it. @wolfv do you have any specific use cases in mine where a list of mirrors would be required/beneficial?

Reply:
We have a long standing goal of making conda-forge more independent from a single vendor. The goal is to achieve something akin to Linux distributions where you have multiple mirrors with telecom providers, universities, etc.

For now there are not many public mirrors (prefix.dev, ghcr.io and beta.mamba.pm, although that is not really up to date afaik). 

Ideally, with time, this would grow. It has the potential to also increase the reliability of the channels in general.

Reply:
E.g. all the mirrors for Fedora: https://admin.fedoraproject.org/mirrormanager/mirrors/Fedora/39/x86_64